### DISCUSSION TITLE
In addition, support Rust plugins (like Nushell)?

### DISCUSSION THREAD

**Discussion by getreu** [helix-editor/helix#9995] [Ideas]
According to [this rationale](https://github.com/helix-editor/helix/discussions/3806) Helix will get a plugin system with an interpreted language serving also for configuration.

I guess the trickiest part is to provide a stable API to hook into Helix internal functions. Why not using this API also for Rust compiled plugins? What would be the benefit of Rust plugins? 

It opens a door to integrate future Rust programmers into Helix' core programmer team, which is -at my opinion- crucial for the long term success of this nice project. 

Technically, the [Nushell plugin system](https://www.nushell.sh/book/plugins.html) system has proven to be very fast and reliable. Nushell's plugin protocol is mature and it's recent update supports streams as well, c.f. [Bidirectional communication and streams for plugins](https://github.com/nushell/nushell/pull/11911).

**Comment by kirawi** [comment]
https://github.com/helix-editor/helix/pull/8675 Steel can load dylibs. The reason Rust plugins were rejected was to make it more accessible to write and use plugins.
